src/OpenTelemetry.Collector.Dependencies/Implementation/AzureSdkDiagnosticListener.cs

```diff
@@ -61,17 +61,28 @@ public override void OnStartActivity(Activity current, object valueValue)
                 }
             }
 
-            List<Link> links = null;
+            List<Link> parentLinks = null;
             if (LinksPropertyFetcher.Fetch(valueValue) is IEnumerable<Activity> activityLinks)
             {
-                links = new List<Link>();
-                foreach (var link in activityLinks)
+                using (var enumerator = activityLinks.GetEnumerator())
                 {
-                    links.Add(new Link(new SpanContext(link.TraceId, link.ParentSpanId, link.ActivityTraceFlags)));
+                    if (enumerator.MoveNext())
+                    {
+                        parentLinks = new List<Link>();
```

Comment:
Is this ugly code what we want so to avoid the allocation of the list when there are no `activityLinks`?

Reply:
I think you could call `activityLinks.Any()` instead - check enumerator is what `Any` does.

Reply:
`.Any()` is a much simpler approach. Just please correct me if I'm wrong but it means 1 more allocation (the second `Enumerator`).

Reply:
GetEnumerator shouldn't allocate https://stackoverflow.com/questions/18552669/memory-allocation-when-using-foreach-loops-in-c-sharp

Anyway, if you used `null` as an indicator of no-links scenario, there wouldn't be even any questions :)

Reply:
lol. Right, but if I call `.Any()` it wouldn't look pretty 😃 

I'll just pass null, wait up